sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -592,6 +594,42 @@ def __init__(self, special_parameter):
                                   check_estimator, MyEstimator)
 
 
+class LRXFailTags(LogisticRegression):
+    def _more_tags(self):
+        return {"_xfail_checks": {
+            "check_complex_data": "This is a bad check"}
+        }
+
+
+def test_check_estimator_xfail_tag_warns():
+    # When a _xfail_checks check passes, warns stating that the test
+    # passes and can be removed from the tag
+
+    assert_warns_message(FutureWarning,
+                         "LRXFailTags:check_complex_data passed, it can "
+                         "be removed from the _xfail_check tag",
+                         check_estimator, LRXFailTags)
+
+
+class LRXDoesNotRaise(LRXFailTags):
+    def fit(self, X, y):
+        # do not raise error for complex check
+        try:
+            return super().fit(X, y)
+        except ValueError as e:
+            if "Complex data not supported" not in str(e):
+                raise
+
+        return self
```

Comment:
Sorry I don't understand what this class and the check below are supposed to test

Also what's the relation with check_estimators_fit_returns_self?

Reply:
The comment needed to be updated to:

```py
# skips check_complex_data based on _xfail_checks
```